File: Go/countwords.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	// Create array of arguments, excluding the first
	args := os.Args[1:]

	words := make(map[string]int)

	for i := 0; i < len(args); i++ {
		arg := strings.ToLower(args[i])

		_, exists := words[arg]
		if exists {
			words[arg]++
		} else {
			words[arg] = 1
		}
	}

	for k, v := range words {
		fmt.Printf("%d %s\n", v, k)
	}
}

File: JavaScript/countwords.js
const words = {};
const args = process.argv;

for (let i = 0; i < args.length; i++) {
  // Making all text lowercase to avoid "The" and "the" being treated as different words
  const word = args[i].toLowerCase();

  if (i == 0 || i == 1) {
    // Ignore the first two command line arguments
    continue;
  } else if (words.hasOwnProperty(word)) {
    // If word exists, increment the value (number of occurences) in the key/value pair
    words[word] = words[word] + 1;
  } else {
    // Add word to map and set the value (number of occurences) to 1s
    words[word] = 1;
  }
}

// Sorting the words by most occurences
const wordsSorted = Object.keys(words)
  .sort(function (a, b) {
    return words[a] + words[b];
  })
  .map((key) => ({ [key]: words[key] }));

// Printing words and occurences to console
for (const word of wordsSorted) {
  for (const k in word) {
    console.log(`${word[k]} ${k}`);
  }
}
